pkgs/checks/lib/src/checks.dart

```diff
@@ -187,14 +186,105 @@ extension ContextExtension<T> on Subject<T> {
   Context<T> get context => _context;
 }
 
-/// The expectation and nesting context already applied to a [Subject].
+/// The context for a [Subject] that allows asserting expectations and creating
+/// nested subjects.
 ///
-/// This is the surface of interaction for expectation extension method
-/// implementations.
+/// Expectation extension methods use [ContextExtension] to get access to the
+/// subject's [Context]. Expectations pass callbacks which describe what is
+/// checked, and perform the expectation, respectively.
```

Comment:
I think I fixed it so that I consistently use "expectation extension method" (or in a few places "expectation extension"), and the term "expectation" should only be used in the more general "a thing that can be expected of a subject" sense.

In the user facing docs describing writing tests, I suspect it will feel natural to describe calling an expectation extension method on a subject as "checking an expectation" on the subject, or for a `ConditionSubject` "recording and expectation".